.github/workflows/check-terraform-syntax.yml

```diff
@@ -27,7 +27,7 @@ jobs:
       - name: Setup terraform
         uses: hashicorp/setup-terraform@v1
         with:
-          terraform_version: 1.0.5  # renovate: datasource=github-releases depName=hashicorp/terraform
+          terraform_version: 1.3.6  # renovate: datasource=github-releases depName=hashicorp/terraform
```

Comment:
I would like to suggest pinning terraform version to [1.4.6](https://releases.hashicorp.com/terraform/1.4.6/)  instead of 1.3.6. 
Two reasons: 
1. 1.4.6 is the latest patch o 1.4 and 1.5 is quite new (June 12) without any patches so far. 
2. I have a real use case for TF 1.4's  `terraform_data` resource feature:
    - I would like to use   `terraform_data`  resource instead of deprecated `null_resource` in https://github.com/SovereignCloudStack/k8s-cluster-api-provider/pull/432 

Then, (probably in a separate PR) I would like to suggest increasing the TF version in [main.tf](https://github.com/SovereignCloudStack/k8s-cluster-api-provider/blob/main/terraform/main.tf#L7) as well. 
IMO the GitHub workflow should test the code with the same version of TF as is defined in `main.tf`.

Reply:
Works for me. Feel free to push 1.4.6 to the branch.